public/framework/catalog_cache/cache_manager.ts

```diff
@@ -68,27 +70,33 @@ export class CatalogCacheManager {
    * Retrieves accelerations cache from local storage.
    * @returns {AccelerationsCacheData} The retrieved accelerations cache.
    */
-  static getAccelerationsCache(): AccelerationsCacheData {
+  static getAccelerationsCache(dataSourceId?: string): AccelerationsCacheData {
     const accelerationCacheData = sessionStorage.getItem(this.accelerationsCacheKey);
 
     if (accelerationCacheData) {
-      return JSON.parse(accelerationCacheData);
-    } else {
-      const defaultCacheObject = {
-        version: CATALOG_CACHE_VERSION,
-        dataSources: [],
-      };
-      this.saveAccelerationsCache(defaultCacheObject);
-      return defaultCacheObject;
+      const cachedAcclerationData = JSON.parse(accelerationCacheData);
+      if (dataSourceId && cachedAcclerationData.client_id === dataSourceId) {
+        return cachedAcclerationData;
+      }
+      else{
+        return cachedAcclerationData
+      }
     }
+    const defaultCacheObject = {
+      version: CATALOG_CACHE_VERSION,
+      client_id: dataSourceId || '' ,
```

Comment:
similar to earlier comment, some are datasource id, some are client id, some are datasourceclientid. maybe better to make them consistent?